src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -6880,6 +6897,25 @@ public void setSendTimeAsDatetime(boolean sendTimeAsDateTimeValue) {
         sendTimeAsDatetime = sendTimeAsDateTimeValue;
     }
 
+    @Override
+    public void setDatetimeParameterType(String datetimeParameterTypeValue) throws SQLServerException {
+        // cast the value to lowercase, so the case in the connection string does not matter
+        if (datetimeParameterTypeValue != null){
+            datetimeParameterTypeValue = datetimeParameterTypeValue.toLowerCase();
+        }
+
+        if (!isValidDatetimeParameterType(datetimeParameterTypeValue)){
+            String errorMessage = "The timestamp encoding value (i.e. " + datetimeParameterTypeValue.toString() + ") must be: datetime, datetime2 or datetimeoffset.";
+            SQLServerException newe = new SQLServerException(errorMessage, null);
+            throw newe;
+        }
+        datetimeParameterType = DatetimeType.valueOfString(datetimeParameterTypeValue);
+    }
+
+    private boolean isValidDatetimeParameterType(String datetimeParameterTypeValue) {
+        return (datetimeParameterTypeValue.equals("datetime") || datetimeParameterTypeValue.equals("datetime2") || datetimeParameterTypeValue.equals("datetimeoffset"));
+    }
```

Comment:
You can run all this through the `DatetimeType.valueOfString(datetimeParameterValueType)` method to validate the input parameter and eliminate the error message instead of hard coding more strings here.

Reply:
this will be updated in a subsequent PR